src/modules/launcher/Plugins/Microsoft.Plugin.Program/Programs/Win32Program.cs

```diff
@@ -403,30 +392,28 @@ private static Win32Program InternetShortcutProgram(string path)
                     {
                         urlPath = line.Substring(urlPrefix.Length);
 
-                        try
-                        {
-                            Uri uri = new Uri(urlPath);
-                        }
-                        catch (UriFormatException e)
+                        if (!Uri.TryCreate(urlPath, UriKind.RelativeOrAbsolute, out Uri _))
                         {
-                            // To catch the exception if the uri cannot be parsed.
-                            // Link to watson crash: https://watsonportal.microsoft.com/Failure?FailureSearchText=5f871ea7-e886-911f-1b31-131f63f6655b
-                            ProgramLogger.Exception($"url could not be parsed", e, MethodBase.GetCurrentMethod().DeclaringType, urlPath);
+                            ProgramLogger.Exception($"url could not be parsed", null, MethodBase.GetCurrentMethod().DeclaringType, urlPath);
```

Comment:
Why wasn't it changed to Warn? What is the reason for changing the second argument from e to null?

Reply:
I've changed The Parse variant to the TryParse variant (so it won't throw an exceptions), this can indeed be changed to a Warn!

Reply:
It is still Exception